src/lib/viewers/controls/color-picker/ColorPickerControl.tsx

```diff
@@ -40,13 +40,23 @@ export default function ColorPickerControl({
 
     const handleClick = (): void => setIsColorPickerToggled(!isColorPickerToggled);
 
+    const handleMouseDown = (event: React.MouseEvent<HTMLButtonElement>): void => {
+        if (event.currentTarget.focus) {
+            // Buttons do not receive focus in Firefox and Safari on MacOS
+            event.currentTarget.focus();
+            // When focus() is called, preventDefault must be called to keep the focus from leaving the target
```

Comment:
`When focus is called within a mousedown handler`, just for clarity?

Reply:
Could this be done in `handleClick` as well?

Reply:
It doesn't affect the current flow, but affects the second click of the button. When you click the button the second time, handle blur happens before handle click, which sets the state to false and back to true again. so you cannot close the palette anymore.

Reply:
The interesting thing here is that even if you click the same button which has the focus, it will still blur and re-focus again, except you handle the logic before everything in `handleMouseDown`